toucan_connectors/snowflake/snowflake_connector.py

```diff
@@ -267,32 +271,41 @@ def _execute_query(self, cursor, query: str, query_parameters: Dict):
         # Prevent error with dict and array values in the parameter object
         query = convert_to_printf_templating_style(query)
         converted_query, ordered_values = convert_to_qmark_paramstyle(query, query_parameters)
+
+        # execute query with timeout.
+        # cf https://docs.snowflake.com/en/user-guide/python-connector-example.html
         query_res = cursor.execute(converted_query, ordered_values)
+        if 'SELECT' in query.upper():
+            if max_rows is None:
+                values = query_res.fetch_pandas_all()
+            else:
+                values = pd.DataFrame.from_dict(query_res.fetchmany(max_rows))
+        else:
+            if max_rows is None:
+                values = pd.DataFrame.from_dict(query_res.fetchall())
+            else:
+                values = pd.DataFrame.from_dict(query_res.fetchmany(max_rows))
+        return values
 
         # https://docs.snowflake.com/en/user-guide/python-connector-api.html#fetch_pandas_all
         # `fetch_pandas_all` will only work with `SELECT` queries, if the
         # query does not contains 'SELECT' then we're defaulting to the usual
         # `fetchall`.
```

Comment:
Ahhh this is the anwser from my question above ^^ Could you move the comment next to the condition plz?

Reply:
:+1: